src/main/resources/amazonsns-config/init.xml

```diff
@@ -38,18 +41,31 @@
         <property name="uri.var.httpRequestUri" value="/"/>
         <property name="DISABLE_CHUNKING" value="true" scope="axis2"/>
         <property name="messageType" expression="get-property('uri.var.contentType')" scope="axis2"/>
+
         <!--
         Need to add port 80 to the 'hostName'property(used to create signature) to match the signature generated by API end.
         Note that ESB sends Host header with port value, hence API uses the receiving value(including port) to generate signature.
+        If the request comes through a proxy, Do not need to add a port.
         -->
-        <filter xpath="get-property('uri.var.enableSSL') = '' or  (not(string(get-property('uri.var.enableSSL'))))">
+        <filter xpath="get-property('uri.var.enableSSL') = '' or  (not(string(get-property('uri.var.enableSSL')))) or get-property('uri.var.enableSSL') = 'false'">
```

Comment:
To check the non existence of the property you can check as following with one condition which will check both conditions (empty or null)
<filter source="boolean(get-property('yourProperty'))" regex="false">
    <then>
           <!-- NULL OR NON EXIST -->
    </then>
    <else>
           <!-- EXIST -->
    </else>
</filter>

Reply:
I tried to fix this problem, but we can' t do it with or get-property('uri.var.enableSSL') = 'false' and getting runtime issue